compat/maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/type/DefaultType.java

```diff
@@ -42,7 +42,7 @@
  * @deprecated since 4.0.0, use {@code maven-api-impl} jar instead
  */
 @Deprecated(since = "4.0.0")
-public class DefaultType implements Type, ArtifactType {
+public class DefaultType implements Type {
```

Comment:
What's the benefit ?

Reply:
The decoupling is a must as we figured, as this class violates (maven or resolver, dependending how you look at it) contract for Type: classifier returns `null` instead of `""` (resolver).